crime-applications-adaptor/api-tests/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/apispecification/RequestSpecificationBuilder.java

```diff
@@ -38,18 +38,28 @@ public class RequestSpecificationBuilder {
   private RequestSpecificationBuilder() {
   }
 
-  public static RequestSpecification getCAMReqSpec() {
-    RequestSpecBuilder requestSpecBuilder = setUpRequestSpecBuilder(CAM_BASE_URL);
+  public static RequestSpecification getCAMCrimeApplyReqSpec() {
+    return getCAMReqSpec(CAM_BASE_URL, CAM_JWT_ISSUER, CAM_JWT_SECRET);
+  }
+
+  public static RequestSpecification getCAMReqSpec(String baseUrl, String jwtIssuer, String jwtSecret) {
+    RequestSpecBuilder requestSpecBuilder = setUpRequestSpecBuilder(baseUrl);
     requestSpecBuilder.addHeader("Authorization",
-        "Bearer " + JwtUtil.generateJwt(CAM_JWT_ISSUER, CAM_JWT_SECRET));
+        "Bearer " + JwtUtil.generateJwt(jwtIssuer, jwtSecret));
     return requestSpecBuilder.build();
   }
 
-  public static RequestSpecification getCAAReqSpec() {
-    RequestSpecBuilder requestSpecBuilder = setUpRequestSpecBuilder(CAA_BASE_URL);
+  public static RequestSpecification getCAACrimeApplyReqSpec() {
+    return getCAAReqSpec(CAA_BASE_URL, CAA_OAUTH_BASE_URL, CAA_OAUTH_CLIENT_ID, CAA_OAUTH_CLIENT_SECRET,
+            CAA_OAUTH_TOKEN_URI);
+  }
+
+  public static RequestSpecification getCAAReqSpec(String baseUrl, String authUrl,
+                                                   String authClientId, String authClientSecret, String authTokenUri) {
+    RequestSpecBuilder requestSpecBuilder = setUpRequestSpecBuilder(baseUrl);
     requestSpecBuilder.addHeader(
         "Authorization", "Bearer " + OAuthTokenUtil.getAccessToken(
-            CAA_OAUTH_BASE_URL, CAA_OAUTH_CLIENT_ID, CAA_OAUTH_CLIENT_SECRET, CAA_OAUTH_TOKEN_URI));
+                    authUrl, authClientId, authClientSecret, authTokenUri));
     return requestSpecBuilder.build();
   }
 
```

Comment:
Probably outside of scope, I would also inline getMaatApiReqSpec, and create a new private method to extract out the duplication of

```
OAuthTokenUtil.getAccessToken(
            CAA_OAUTH_BASE_URL, CAA_OAUTH_CLIENT_ID, CAA_OAUTH_CLIENT_SECRET, CAA_OAUTH_TOKEN_URI)
```

Reply:
I've extracted that bit to a private method. Regarding inlining getMattApiReqSpec, see my previous comment above :)